app/(auth)/confirm-email/page.tsx

```diff
@@ -0,0 +1,26 @@
+import { MailCheck as ConfirmEmailIcon } from "lucide-react";
+
+export default function ConfirmEmailPage({
+  searchParams,
+}: {
+  searchParams?: { [key: string]: string | string[] | undefined };
```

Comment:
It is better to define the type explicitly, since It's only 1 param,
```suggestion
  searchParams?: { email: string };
```